### ISSUE TITLE
Remove host and port from URLs exposed in Bigchaindb http endpoints

### ISSUE THREAD

**Issue by krish7919** [bigchaindb/bigchaindb#1141]
We should remove all references to the protocol, host and the port at which BigchainDB is available.
For one, the users already know about it if they can hit the root URL.
Secondly, BigchainDB will not be aware where is it running in the future - it might be frontended by a load balancer, a firewall or pretty much anything else.

Planning to remove all references to `http://localhost:59984` in the following output:
```
curl -X GET http://localhost:59984/
{
  "_links": {
    "api_v1": "http://localhost:59984/api/v1/", 
    "docs": "https://docs.bigchaindb.com/projects/server/en/0.9.dev/"
  }, 
  "keyring": [], 
  "public_key": "6F5Bi9HEgaFicPiT1FNDCPJgQBhAr8SJVUX7P2uBKLRY", 
  "software": "BigchainDB", 
  "version": "0.9.0.dev"
}
curl -X GET http://localhost:59984/api/v1
{
	"_links": {
		"docs": "https://docs.bigchaindb.com/projects/server/en/0.9.dev/drivers-clients/http-client-server-api.html",
		"statuses": "http://localhost:59984/api/v1/statuses/",
		"transactions": "http://localhost:59984/api/v1/transactions/",
		"self": "http://localhost:59984/api/v1/"
	}
}
```

**Comment by TimDaub** [comment]
For the completion of this ticket, it is also necessary to handle this for the Events API specification and implementation.

**Comment by TimDaub** [comment]
Consider that node operators will want to set relative paths (or as @krish7919 called it "prefixes") for `api_v1`. Description here: https://github.com/bigchaindb/bigchaindb/issues/1141

**Comment by krish7919** [comment]
@TimDaub 
To reiterate, the proposal suggests stripping all the `protocol://host:port` from all the exposed URLs from our root endpoint.
I do not get the point of configuring them. The users are already aware of the information. It is overhead on our side if we go with configuring these details, not to mention the chance of a security risk if the user hits `https://host1:port1/` and the return value has `"api_v1": "https://host2:port2/api/v1/"`.

**Comment by TimDaub** [comment]
Moved to "BDB 1.0: Breaking HTTP API changes, 2nd iteration HTTP API" milestone. Will handle in #1522

**Comment by TimDaub** [comment]
As was decided with the team in #1522: We'll not go for a configurable option, but simply remove host+port from all links in `/` and `/api/v1`, as outlined [here](https://github.com/bigchaindb/bigchaindb/blob/99499b1f8783719a082813912ac9a0d363ae278f/bdb-ip.md#proposal-2-4).

**Comment by TimDaub** [comment]
Fixed via #1538
